### PR TITLE
Users should be able to search other users and manage their own profiles

### PR DESCRIPTION
**Pull request by azaria-dedmon**
**What does this PR do?**

- This PR provides users the functionality to search other users within the system and also edit/delete their profiles.

**Description of task to be completed.**

- Provides functionality for user's to edit/delete their own profiles in `init.py`
- Provides functionality for users to search other users in `init.py`
- Includes relevant unit tests.

**How should this be manually tested?**

- Run `Python3 run.py`
- Go to `localhost:5000/`
- Click the register link
- After signup, search for users by username in the search bar on the `user home page`
- Click the edit link on the `user home page` to edit profile information.
- Click the edit link and then click the delete button to process account deletion.

### CODE REVIEW

**Comment by alexkayabula** [comment]
@azaria-dedmon why are we having two commits for this pull request ?

**Comment by alexkayabula** [comment]
The title of this PR is confusing, does this mean you can edit another user's profile.

**Comment by alexkayabula** [comment]
The description of the tasks is so wordy, listing the tasks is not necessary.

File: app/__init__.py [hunk 1]
```diff
@@ -109,4 +111,51 @@ def show_state_locations():
...
+        g.user.lastname = form.lastname.data
+        g.user.username = form.username.data
+        g.user.email = form.email.data
+        g.user.image = form.image.data
+        g.user.state = form.state.data
+        g.user.vax_date = form.vax_date.data
+        g.user.covid_status = form.covid_status.data
+        db.session.commit()
+        return redirect('/user')
+    return render_template('users/user_edit.html', form=form)
```
Comment (alexkayabula) [inline app/__init__.py:131]: @azaria-dedmon this should be in a PR of its own.

File: app/__init__.py [hunk 2]
```diff
@@ -109,4 +111,51 @@ def show_state_locations():
...
+        if bcrypt.check_password_hash(g.user.password, form.password.data):
+            do_logout()
+            db.session.delete(g.user)
+            db.session.commit()
+            return redirect('/')
+        else:
+            flash("Invalid Password", "danger")
+            return render_template('users/user_delete.html', form=form)
+
+    return render_template('users/user_delete.html', form=form)
```
Comment (alexkayabula) [inline app/__init__.py:149]: This should be in a PR of its own.

File: app/__init__.py [hunk 3]
```diff
@@ -109,4 +111,51 @@ def show_state_locations():
...
+
+@app.route('/search-user')
+def get_searched_user():
+    """Search other users"""
+
+    username = request.args.get('username')
+    searched_user =  User.query.filter_by(username=username).first()
+
+    return render_template('users/searched_user.html', searched_user=searched_user)
```
Comment (alexkayabula) [inline app/__init__.py:158]: This should be in a PR of its own.

**Review by alexkayabula — CHANGES_REQUESTED** [review]

**Comment by alexkayabula** [comment]
In this project, will any registered user be able to delete or edit other users profiles ?

**Comment by azaria-dedmon** [comment]
> In this project, will any registered user be able to delete or edit other users profiles ?

No I had accidentally miss worded the title. I meant, "Users should be able to search other users and manage their _own_ profile".

**Pull request closed without merging**